server/src/test/java/io/druid/segment/loading/LoadSpecTest.java

```diff
@@ -0,0 +1,143 @@
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.InjectableValues;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Binder;
+import com.google.inject.Injector;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.metamx.common.IAE;
+import io.druid.guice.GuiceInjectors;
+import io.druid.jackson.DefaultObjectMapper;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/**
+ *
+ */
+@RunWith(Parameterized.class)
+public class LoadSpecTest
+{
+  @Parameterized.Parameters
+  public static Collection<Object[]> getParameters()
+  {
+    return ImmutableList.<Object[]>of(
+        new Object[]{"{\"path\":\"/\",\"type\":\"local\"}", "local"}
+    );
+  }
+
+  private final String value;
+  private final String expectedId;
+
+  public LoadSpecTest(String value, String expectedId)
+  {
+    this.value = value;
+    this.expectedId = expectedId;
+  }
+
+  private static ObjectMapper mapper;
+
+  @BeforeClass
+  public static void setUp()
+  {
+    final Injector injector = GuiceInjectors.makeStartupInjectorWithModules(
+        ImmutableList.of(
+            new Module()
+            {
+              @Override
+              public void configure(Binder binder)
+              {
+                binder.bind(LocalDataSegmentPuller.class);
+              }
+            }
+        )
+    );
+    mapper = new DefaultObjectMapper();
+    mapper.registerModule(
+        new com.fasterxml.jackson.databind.Module()
+        {
+          @Override
+          public String getModuleName()
+          {
+            return "noName";
+          }
+
+          @Override
+          public Version version()
+          {
+            return Version.unknownVersion();
+          }
+
+          @Override
+          public void setupModule(SetupContext context)
+          {
+            context.registerSubtypes(LocalLoadSpec.class);
+          }
+        }
+    );
+    mapper.setInjectableValues(
+        new InjectableValues()
+        {
+          @Override
+          public Object findInjectableValue(
+              Object valueId, DeserializationContext ctxt, BeanProperty forProperty, Object beanInstance
+          )
+          {
+            // From the docs:   "Object that identifies value to inject; may be a simple name or more complex identifier object,
+            //                  whatever provider needs"
+            // Currently we should only be dealing with `Key` instances, and anything more advanced should be handled with
+            // great care
+            if (valueId instanceof Key) {
+              return injector.getInstance((Key) valueId);
+            }
+            if (valueId instanceof String) {
```

Comment:
The reason you are getting a String here is because you don't have the GuiceAnnotationIntrospector registered with the ObjectMapper.